[webkitgtk 2.44.0] fix compilation if !ENABLE(VIDEO)

ThomasDevoogdt opened the pull request:
Bug report:
  - https://bugs.webkit.org/show_bug.cgi?id=271881.

PR to main:
  - https://github.com/WebKit/WebKit/pull/26651

<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/013083f09c892b912553969bc3f11831f446e463

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46373 "18 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25513 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48966 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/49048 "Unable to build WebKit without PR, please check manually (failure)") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/42414 "The change is no longer eligible for processing.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/48680 "Passed tests") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29872 "Unable to build WebKit without PR, please check manually (failure)") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Sonoma-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/49048 "Unable to build WebKit without PR, please check manually (failure)") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46951 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/49/builds/29872 "Unable to build WebKit without PR, please check manually (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39982 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/49048 "Unable to build WebKit without PR, please check manually (failure)") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/49/builds/29872 "Unable to build WebKit without PR, please check manually (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41087 "Passed tests") | [❌ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4418 "Unable to build WebKit without PR, please check manually (failure)") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/49/builds/29872 "Unable to build WebKit without PR, please check manually (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41451 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50874 "Unable to build WebKit without PR, please check manually (failure)") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/21378 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Sonoma-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/50874 "Unable to build WebKit without PR, please check manually (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22669 "Built successfully") | | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/50874 "Unable to build WebKit without PR, please check manually (failure)") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/23039 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22368 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski reviewed (commented):
Seems fine once Linux is building

ThomasDevoogdt committed:
Cherry-pick 276977@main (ea2be408f27e).

https://bugs.webkit.org/show_bug.cgi?id=271881

    REGRESSION: fix compilation if !ENABLE(VIDEO)

    https://bugs.webkit.org/show_bug.cgi?id=271881

    Reviewed by Mike Wyrzykowski.

    [WebCore]: FullscreenManager: add missing include

    HTMLMediaElement normally includes HTMLMediaElementEnums,
    but only if VIDEO is enabled.

    [WebCore]: FullscreenManager: fix undefined reference to HTMLMediaElement

    HTMLMediaElement is not compiled if VIDEO is not enabled.

    [WebCore]: css: fix fix undefined reference to HTMLMediaElement

    HTMLMediaElement is not compiled if VIDEO is not enabled.

    [WebCore]: InspectorDOMAgent: fix fix undefined reference to HTMLMediaElement

    HTMLMediaElement is not compiled if VIDEO is not enabled.

    [WebKit]: WebGPU: drop duplicate directive in RemoteDevice

    The #if PLATFORM(COCOA) && ENABLE(VIDEO) directive is already defined
    outside the function scope, so drop the inner ones.

    [WebCore]: GPUDevice: fix undefined reference to HTMLMediaElement

    HTMLMediaElement is not compiled if VIDEO is not enabled.

    [WebKit]: WebFullScreenManagerProxy: add missing include

    HTMLMediaElement normally includes HTMLMediaElementEnums,
    but only if VIDEO is enabled.

    [WebKit]: WebFullScreenManager: add missing include

    HTMLMediaElement normally includes HTMLMediaElementEnums,
    but only if VIDEO is enabled.

    [WebKit]: InjectedBundlePageFullScreenClient: add missing include

    HTMLMediaElement normally includes HTMLMediaElementEnums,
    but only if VIDEO is enabled.

    [WebKit]: WebCoreArgumentCoders.serialization.in: fix undefined reference

    CaptionUserPreferencesDisplayMode and SerializedPlatformDataCueValue
    are only compiled if VIDEO is enabled.

    Signed-off-by: Thomas Devoogdt <thomas@devoogdt.com>
    Canonical link: https://commits.webkit.org/276977@main

(cherry picked from commit ea2be408f27eb1009bbde7b0a2496018d4ca5e86)
Signed-off-by: Thomas Devoogdt <thomas@devoogdt.com>
